notebooks/NIRISS_WFSS_advanced/01_niriss_wfss_image2_image3.ipynb

```diff
@@ -0,0 +1,1070 @@
+    "  - [Matching Source IDs Across Catalogs](#match_sources) \n",
+    "  - [Manually Editing the Source Catalog](#manual_cat)\n",
+    "\n",
+    "**Author**: Rachel Plesha (rplesha@stsci.edu), Camilla Pacifici (cpacifici@stsci.edu), JWebbinar notebooks.<br>\n",
+    "**First Published**: January 2024"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7fdaf1a1-57e5-4239-b82e-6552261d8d82",
+   "metadata": {},
+   "source": [
+    "<a id='imports'></a>\n",
+    "## Imports & Data Setup"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e3ccd6a3-e781-4d7f-8120-fa142723a6d9",
+   "metadata": {},
+   "source": [
+    "[CRDS Documentation](https://jwst-pipeline.readthedocs.io/en/latest/jwst/user_documentation/reference_files_crds.html#crds)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "6bf2ea52-2568-44cf-8074-eec3b5b76cc0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Update the CRDS path to your local directory\n",
+    "%env CRDS_PATH=crds_cache\n",
+    "%env CRDS_SERVER_URL=https://jwst-crds.stsci.edu"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "028352f4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import glob\n",
+    "import json\n",
+    "import warnings\n",
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "\n",
+    "import astropy.units as u\n",
+    "from astropy.io import fits\n",
+    "from astropy.coordinates import SkyCoord\n",
+    "from astropy.table import Table\n",
+    "\n",
+    "from jdaviz import Imviz\n",
+    "from matplotlib import pyplot as plt\n",
+    "%matplotlib inline\n",
+    "\n",
+    "import jwst\n",
+    "from jwst.pipeline import Image2Pipeline\n",
+    "from jwst.pipeline import Image3Pipeline"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "fb035a4d-e63c-4125-b083-462e3353e49f",
+   "metadata": {},
+   "source": [
+    "Check what version of the JWST pipeline you are using. To see what the latest version of the pipeline is available, check <a href=\"https://github.com/spacetelescope/jwst#software-vs-dms-build-version-map\">here</a>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ae10a560",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print('jwst:', jwst.__version__)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "feb25e11-bc16-44b8-830c-d34f12e2dcf7",
+   "metadata": {},
+   "source": [
+    "The data directory, `data_dir` here should contain all of the association and rate files in a single, flat directory. `default_run_image3` and `custom_run_image3` are directories that we will use later for our calibrated data. They are separated so that we can compare the two outputs."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e2ecc213-479a-422c-a0f8-8608309b845c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "data_dir = 'data'\n",
```

Comment:
Somewhere around here, you can download and expand the zipped file with all the rate and association files: https://data.science.stsci.edu/redirect/JWST/jwst-data_analysis_tools/niriss_wfss_advanced/niriss_wfss_advanced_01_input.zip

Reply:
@camipacifici is it necessary to have this in a cell that is run by default for the checks to run? I'd rather put this in a "raw" cell since it can take several minutes to download, and the 1st notebook in this series does that for them.

Reply:
They need to be run to pass the pep8 anyway, so I just have them as normal code blocks now.

Reply:
Yea, the notebooks do not talk to each other, so you can expect that CI does NOT use the output from notebook 1 when testing notebook 2.